### PR TITLE
Fix optionalReplacementSpan not being applied to completion entries

### PR DESCRIPTION
**Pull request by johnsoncodehk**
This will fix the problem with the first video shown at https://github.com/vuejs/language-tools/discussions/3789#discussioncomment-7815181.

If TS CompletionEntry has no replacementSpan, VSCode will fallback to the wordPattern of the language configuration, but for Vue documents, VSCode will not use the wordPattern of the embedded language, but the wordPattern of the vue language, so the `$xxx` variable will not be replaced correctly.

### CODE REVIEW

**Review by mjbvz — DISMISSED** [review]

**Comment by amunger** [comment]
formatting error in the linter?

File: extensions/typescript-language-features/src/languageFeatures/completions.ts [hunk 1]
```diff
@@ -760,6 +760,12 @@ class TypeScriptCompletionItemProvider implements vscode.CompletionItemProvider<
 			isIncomplete = !!response.body.isIncomplete || (response as any).metadata && (response as any).metadata.isIncomplete;
 			entries = response.body.entries;
 			metadata = response.metadata;
+			
+			if (response.body.optionalReplacementSpan) {
+				for (const entry of entries) {
+					entry.replacementSpan ??= response.body.optionalReplacementSpan;
```
Comment (rzhao271) [inline extensions/typescript-language-features/src/languageFeatures/completions.ts:766]: Sanity check: are there per-entry optional replacement spans, or is there only a single one to use for all of the entries?

Comment (johnsoncodehk) [reply extensions/typescript-language-features/src/languageFeatures/completions.ts:766]: It's a single one to use for all of the entries, see https://github.com/microsoft/TypeScript/blob/a36d04fc63c83b6ec5a8099942b653a5caa29eb1/src/services/types.ts#L1382-L1384

**Review by rzhao271 — APPROVED** [review]

**Comment by johnsoncodehk** [comment]
> formatting error in the linter?

It seems to be caused by my branch not being up to date and merging branch on CI, now it should fixed.

**Review by joyceerhl — APPROVED** [review]

**Comment by johnsoncodehk** [comment]
@mjbvz auto-merge has been suspended by comments, can you enable it again? Thanks!

**Review by mjbvz — APPROVED** [review]

**Pull request merged**